### PR TITLE
Remove params/attributes from API reference TOC

### LINKED ISSUE #361: Remove parameters from table of contents in API reference
#### Is your feature request related to a problem? Please describe
It's somewhat hard to navigate the API reference, especially https://pyvrp.org/api/pyvrp.html. Part of this is due to the table of contents being cluttered with parameters.

#### Describe the solution you'd like
I propose to remove parameters from table of contents. I think that already improves the navigation a lot.

### PR DESCRIPTION
**Pull request by leonlan**
This PR:

- [x] Closes #361.
- [x] Adds and passes relevant tests.
- [ ] Has well-formatted code and documentation.

<details>

**Notes**:

- It is essential that you add a test when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: docs/source/api/search.rst [hunk 1]
```diff
@@ -15,11 +15,13 @@ All search methods implement the :class:`SearchMethod` protocol.
 
    .. autoclass:: SearchMethod
       :members:
+      :special-members: __call___
```
Comment (leonlan) [inline docs/source/api/search.rst:18]: Strange, this doesn't work for the `search` functions but it does work for `StoppingCriterion`.

**Comment by N-Wouda** [comment]
> Parameters do not nicely cross-reference anymore: `numpydoc` isn't as good as `napoleon` for that. But the TOC can now be navigated much easier, which I think is more important.

Yeah, that's why I want to keep using `napoleon`. The problem you're trying to fix isn't solved by this, it's just that `numpydoc` doesn't seem to understand the code all that well. [This](https://jbms.github.io/sphinx-immaterial/apidoc/index.html#object-description-options) is a far more robust solution. I think the relevant key is `py.*Param`?

**Comment by leonlan** [comment]
> > Parameters do not nicely cross-reference anymore: `numpydoc` isn't as good as `napoleon` for that. But the TOC can now be navigated much easier, which I think is more important.
> 
> Yeah, that's why I want to keep using `napoleon`. The problem you're trying to fix isn't solved by this, it's just that `numpydoc` doesn't seem to understand the code all that well. [This](https://jbms.github.io/sphinx-immaterial/apidoc/index.html#object-description-options) is a far more robust solution. I think the relevant key is `py.*Param`? Possibly also need to remove attributes, which I guess would be `py.*Attribute`.

I'll have a look!

**Review by N-Wouda — APPROVED** [review]

**Comment by N-Wouda** [comment]
It might be possible to shorten some part using `napoleon_use_ivar = True`, but I don't know exactly what that does. I'll have a look locally.

**Comment by N-Wouda** [comment]
OK nvm, this is good as is.

**Pull request merged**